experimenter/experimenter/nimbus_ui_new/forms.py

```diff
@@ -535,6 +539,19 @@ def save(self, commit=True):
         experiment.status_next = self.status_next
         experiment.publish_status = self.publish_status
         experiment.save()
+        # Allocate bucket range if needed
+        if experiment.has_filter(NimbusExperiment.Filters.SHOULD_ALLOCATE_BUCKETS):
+            experiment.allocate_bucket_range()
+
+        if experiment.should_call_preview_task:
+            nimbus_synchronize_preview_experiments_in_kinto.apply_async(countdown=5)
+
+        if experiment.should_call_push_task:
+            collection = experiment.kinto_collection
+            nimbus_check_kinto_push_queue_by_collection.apply_async(
+                countdown=5, args=[collection]
+            )
```

Comment:
Oh so yes this mirrors the way we did it in the serializer where it's a single save method handling all possible state changes.  But I think we can simplify this even more since we have specific state transition forms below, so rather than checking whether we're in some state transition and whether we should do any of these, we can just hard code them into the save methods of the appropriate forms below, and then we don't need to check which state transition we're in.